description: support build flags passed in the --args to gn
author: Michael Gilbert <mgilbert@debian.org>

--- a/build/toolchain/linux/BUILD.gn
+++ b/build/toolchain/linux/BUILD.gn
@@ -5,6 +5,12 @@
 import("//build/config/sysroot.gni")
 import("//build/toolchain/gcc_toolchain.gni")
 
+declare_args() {
+  target_extra_cflags = ""
+  target_extra_ldflags = ""
+  target_extra_cxxflags = ""
+}
+
 clang_toolchain("clang_arm") {
   toolprefix = "arm-linux-gnueabihf-"
   toolchain_args = {
@@ -50,6 +56,10 @@ gcc_toolchain("arm") {
   readelf = "${toolprefix}readelf"
   nm = "${toolprefix}nm"
 
+  extra_cflags = target_extra_cflags
+  extra_ldflags = target_extra_ldflags
+  extra_cxxflags = target_extra_cxxflags
+
   toolchain_args = {
     current_cpu = "arm"
     current_os = "linux"
@@ -89,6 +99,10 @@ gcc_toolchain("x86") {
   ar = "ar"
   ld = cxx
 
+  extra_cflags = target_extra_cflags
+  extra_ldflags = target_extra_ldflags
+  extra_cxxflags = target_extra_cxxflags
+
   toolchain_args = {
     current_cpu = "x86"
     current_os = "linux"
@@ -128,6 +142,10 @@ gcc_toolchain("x64") {
   ar = "ar"
   ld = cxx
 
+  extra_cflags = target_extra_cflags
+  extra_ldflags = target_extra_ldflags
+  extra_cxxflags = target_extra_cxxflags
+
   toolchain_args = {
     current_cpu = "x64"
     current_os = "linux"
